cvmfs/util/shared_ptr.h

```diff
@@ -71,8 +71,8 @@ class SharedPtr {
   template <class Y>
   SharedPtr& operator=(SharedPtr<Y> const& r) {  // never throws
     Reset();
-    value_ = r.value_;
-    count_ = r.count_;
+    value_ = r.Get();
+    count_ = r.GetCountPtr();
```

Comment:
because these are public method of a shared pointer that might belong to different classes. (`<T>` vs `<Y>`) so they cannot access the private members. 
Example usage:
```c++
case kShardingPolicyExternal: {
    SharedPtr<LoadBalancing> externalPolicy =
                        SharedPtr<LoadBalancing>(new LoadBalancing());
    health_check_ = externalPolicy;  // shared_ptr
    sharding_policy_ = externalPolicy;  //shared_ptr
    externalPolicy.Reset();
    success = true;
} break;
```

Reply:
(I didn't see that the template arguments are different)